grc/core/Platform.py

```diff
@@ -158,7 +158,12 @@ def build_block_library(self):
 
         # Add blocks to block tree
         for key, block in self.blocks.iteritems():
-            category = self._block_categories.get(key, block.category)
+            category = block.category
+            if len(category) == 0:
+                category = self._block_categories.get(key, block.category)
+            elif key in self._block_categories and self._block_categories[key] != category:
+                print >> sys.stderr, 'conflicting categories on %s:\n\t"%s" (block file) vs "%s" (block tree)' % \
+                (key, '/'.join(category), '/'.join(self._block_categories[key]))
```

Comment:
Because you mentioned it: `elif self._block_categories.get(key, None) is not None:` is standard Pythonesque (and easier on the eye).
You even might leave out the `is not None` part since `None` is never `True`.